man/tpm2_hmac.1.md

```diff
@@ -0,0 +1,75 @@
+tpm2_hmac 1 "SEPTEMBER 2017" tpm2-tools
+==================================================
+
+NAME
+----
+
+tpm2_hmac(1) - Performs an HMAC operation with the TPM.
+
+SYNOPSIS
+--------
+
+`tpm2_hmac` [OPTIONS] _FILE_
+
+DESCRIPTION
+-----------
+
+tpm2_hmac(1) - performs an HMAC operation on _FILE_ and returns the results. If
+_FILE_ is not specified, then data is read from stdin.
+
+OPTIONS
+-------
+
+ * `-k`, `--keyHandle`=_KEY\_CONTEXT\_FILE_:
+    The key handle for the symmetric signing key providing the HMAC key.
+
+  * `-c`, `--keyContext`=_KEY\_CONTEXT\_FILE_:
+    The filename of the key context used for the operation.
+
+  * `-P`, `--pwdk`=_KEY\_PASSWORD_:
+    The password for key, optional. Passwords should follow the
+    "password formatting standards, see section "Password Formatting".
+
+  * `-g`, `--halg`=_HASH\_ALGORITHM_:
+    The hash algorithm to use.
+    Algorithms should follow the "formatting standards, see section
+    "Algorithm Specifiers".
+    Also, see section "Supported Hash Algorithms" for a list of supported hash
+    algorithms.
+
+  * `-o`, `--outfile`=_OUT\_FILE_
+    Optional file record of the HMAC result. Defaults to stdout.
+
+  * `-S`, `--input-session-handle`=_SESSION\_HANDLE_:
+    Optional Input session handle from a policy session for authorization.
+
+[common options](common/options.md)
+
+[common tcti options](common/tcti.md)
+
+[password formatting](common/password.md)
+
+[supported hash algorithms](common/hash.md)
+
+[algorithm specifiers](common/alg.md)
+
+EXAMPLES
+--------
+
+```
+tpm2_hmac  -k 0x81010002 -P abc123 -g 0x004 -I <inputFilename> -o <hmacFilename>
+tpm2_hmac  -c key.context -P abc123 -g 0x004 -I <inputFilename> -o <hmacFilename>
+tpm2_hmac  -k 0x81010002 -P 123abc -X -g 0x004 -I <inputFilename> -o <hmacFilename>```
```

Comment:
Here too, using pretty names in the examples makes it easier for users.

Reply:
I can definitively drop the -X options and use friendly names.